backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+from .models import News
+from rest_framework import viewsets
+from .serializers import NewsSerializer
+
+#This is an example of filtering the pieces of news with the city-id like this: /?city-id=5
+class FilterByCity(object):   
+    def get_queryset(self):   #You need to override the default get_queryset method
+        queryset = super().get_queryset()
+        city_id = self.request.query_params.get('city-id')
+        if city_id is not None:
+            queryset = queryset.filter(city=city_id)
+        return queryset
+
+
+class NewsViewSet(FilterByCity, viewsets. ):
+    queryset = News.objects.all()
+    serializer_class = NewsSerializer
+```
+
+- The NewsViewSet class will now inherit both classes FilterByCity viewsets.ModelViewSet from right to left. The FilterByCity get_queryset method will override the ModelViewSet one because the class is the furthest at the left.
+
+
+
+### Create a News ViewSet *(~2 min)*
+
+- Create a route for the ViewSet.
+
+ouproject/config/router.py
+```python
+from rest_framework import routers
+from intramuros.users.viewsets import UserViewSet
+from intramuros.locations.viewsets import CityViewSet
+from intramuros.publications.viewsets import NewsViewSet  #Add this line
+
+
+# Routers provide an easy way of automatically determining the URL conf.
+router = routers.DefaultRouter()
+router.register(r'users', UserViewSet)
+router.register(r'locations/cities', CityViewSet)
+router.register(r'publications/news', NewsViewSet)    #Add this one too
+```
+
+- You are good to go! :D
+
+- If you go on this url: HOST/publications/news/
+
+- you should get something like this:
+
+```json
+[
+    {
+        "title": "Reading the article",
+        "description": "Hello, I am glad I read this article!",
+        "created_at": "2017-11-08T15:31:30.597524Z",
+        "updated_at": "2017-11-08T15:31:30.597555Z",
+        "city": "66",
+        "id": 2
+    },
+    {
+        "title": "Hate comment",
+        "description": "I didn't find this interesting, I already knew everything...",
+        "created_at": "2017-11-08T15:33:28.834694Z",
+        "updated_at": "2017-11-08T15:33:28.834721Z",
+        "city": "66",
+        "id": 3
+    },
+    {
+        "title": "Les français adorent !",
+        "description": "Je suis française et j'ai lu cet article, il m'a aidé à faire mon premier modèle, youhouu !",
+        "created_at": "2017-11-08T15:20:33.457740Z",
+        "updated_at": "2017-11-08T16:54:19.322703Z",
+        "city": "5",
+        "id": 1
+    }
+]
+```
+
+
+- If you go on this url: HOST/publications/news/?city-id=5 (that is filtered)
+
+- you should get something like this:
+
+```json
+[
+    {
+        "title": "Les français adorent !",
+        "description": "Je suis française et j'ai lu cet article, il m'a aidé à faire mon premier modèle, youhouu !",
+        "created_at": "2017-11-08T15:20:33.457740Z",
+        "updated_at": "2017-11-08T16:54:19.322703Z",
+        "city": "5",
+        "id": 1
+    }
+]
+```
```

Comment:
Super cool to share this in the dev standards! Thanks @AliceB08

Reply:
Really nice but needs some more what and why